### PR TITLE
SDMEXT-913:Multitenancy Changes in the plugin

### PR DESCRIPTION
**Pull request by rashmiangadi11**
https://jira.tools.sap/browse/SDMEXT-878

### CODE REVIEW

**Comment by yashmeet29** [comment]
Please add the Jira link in description

File: sdm/src/main/java/com/sap/cds/sdm/caching/RepoKey.java
Comment (yashmeet29) [inline sdm/src/main/java/com/sap/cds/sdm/caching/RepoKey.java:1]: Please add this file in sonar exclusion in pom.xml

File: sdm/src/test/java/com/sap/cds/sdm/handler/TokenHandlerTest.java [hunk 1]
```diff
@@ -288,44 +288,42 @@ public void testGetAccessToken() {
...
     SDMCredentials mockSdmCredentials = Mockito.mock(SDMCredentials.class);
     when(mockSdmCredentials.getClientId()).thenReturn("mockClientId");
     when(mockSdmCredentials.getClientSecret()).thenReturn("mockClientSecret");
     when(mockSdmCredentials.getBaseTokenUrl()).thenReturn("https://mock.url");
 
-    Cache<String, String> mockCache = Mockito.mock(Cache.class);
-    when(mockCache.get(any())).thenReturn(null); // Cache is empty
-
+    String token =
+        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImpvaG4uZG9lQGV4YW1wbGUuY29tIiwic3ViIjoiMTIzNDU2Nzg5MCIsIm5hbWUiOiJKb2huIERvZSIsImlhdCI6MTY4MzQxODI4MCwiZXhwIjoxNjg1OTQ0MjgwLCJleHRfYXR0ciI6eyJ6ZG4iOiJ0ZW5hbnQifX0.efgtgCjF7bxG2kEgYbkTObovuZN5YQP5t7yr9aPKntk";
```
Comment (smaheshbhat) [inline sdm/src/test/java/com/sap/cds/sdm/handler/TokenHandlerTest.java:298]: Shall we refrain from hardcoding the token ? This is for test and will be expired key anyway. So why not just have placeholder string instead ?

Comment (rashmiangadi11) [reply sdm/src/test/java/com/sap/cds/sdm/handler/TokenHandlerTest.java:298]: we decode the token and fetch some details and need the hard coded token for the same

**Review by yashmeet29 — APPROVED** [review]

**Pull request merged**